tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/polyparser/ArcEagerTransitionSystem.scala

```diff
@@ -500,3 +281,86 @@ class ArcEagerGuidedCostFunction(
   }
 }
 
+/** The ArcEagerForbiddenArcInterpretation handles ForbiddenArc constraints for the arc-eager
+  * system. In other words, it translates these constraints into a function that returns true
+  * for any (state, transition) pair that violates the constraint.
+  *
+  * @param forbiddenArc the forbidden arc constraint to consider
+  */
+case class ArcEagerForbiddenArcInterpretation(
+    forbiddenArc: ForbiddenEdge
+) extends ParsingConstraintInterpretation {
+
+  def applyToParserState(state: TransitionParserState, transition: StateTransition): Boolean = {
+    state match {
+      case tpState: TransitionParserState =>
```

Comment:
When is `tpState` ever not a `TransitionParserState`? It says it is one right in the function signature.